client/peerset/src/lib.rs

```diff
@@ -160,26 +176,36 @@ impl From<u64> for IncomingIndex {
 /// Configuration to pass when creating the peer set manager.
 #[derive(Debug)]
 pub struct PeersetConfig {
+	/// List of sets of nodes the peerset manages.
+	pub sets: Vec<SetConfig>,
+
+	/// If true, we only accept nodes in [`SetConfig::reserved_nodes`] of any of the sets.
+	pub reserved_only: bool,
+}
+
+/// Configuration for a single set of nodes.
+#[derive(Debug)]
+pub struct SetConfig {
+	/// Name of the set. Used to later refer to that set.
+	pub name: &'static str,
```

Comment:
Do I understand correctly that set names have to be known at compile time?

Does Polkadot know all set names at compile time (//CC @ordian)? E.g. when a validator is assigned to a specific parachain, would it use a prefixed priority group or reuse the same from the previous assignment?

Reply:
This is indeed one of the aspects I'm hesitating about. To me it feels more correct to indicate ahead of time the list of overlay networks, and Polkadot would indeed use a set called something like "current-validators-rotation".

Reply:
> To me it feels more correct to indicate ahead of time the list of overlay networks

Otherwise, there isn't really a clean way to configure the number of slots of each set.

Reply:
> Do I understand correctly that set names have to be known at compile time?
> 
> Does Polkadot know all set names at compile time (//CC @ordian)? E.g. when a validator is assigned to a specific parachain, would it use a prefixed priority group or reuse the same from the previous assignment?

That's a good question, I don't think we have a design for it now. It depends on how peer-sets are working, but for now we have one large group that is changing over time.

Reply:
The intention behind this refactor is to use one "peers set" and change its content.